sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/MutationGroupEncoderTest.java

```diff
@@ -0,0 +1,270 @@
+
+/**
+ * Tests for {@link MutationGroupEncoder}.
+ */
+public class MutationGroupEncoderTest {
+  private SpannerSchema allTypesSchema;
+
+  @Before
+  public void setUp() throws Exception {
+    SpannerSchema.Builder builder = SpannerSchema.builder();
+
+    builder.addColumn("test", "intkey", "INT64");
+    builder.addKeyPart("test", "intkey", false);
+
+    builder.addColumn("test", "bool", "BOOL");
+    builder.addColumn("test", "int64", "INT64");
+    builder.addColumn("test", "float64", "FLOAT64");
+    builder.addColumn("test", "string", "STRING");
+    builder.addColumn("test", "bytes", "BYTES");
+    builder.addColumn("test", "timestamp", "TIMESTAMP");
+    builder.addColumn("test", "date", "DATE");
+
+    builder.addColumn("test", "nullbool", "BOOL");
+    builder.addColumn("test", "nullint64", "INT64");
+    builder.addColumn("test", "nullfloat64", "FLOAT64");
+    builder.addColumn("test", "nullstring", "STRING");
+    builder.addColumn("test", "nullbytes", "BYTES");
+    builder.addColumn("test", "nulltimestamp", "TIMESTAMP");
+    builder.addColumn("test", "nulldate", "DATE");
+
+    builder.addColumn("test", "arrbool", "ARRAY<BOOL>");
+    builder.addColumn("test", "arrint64", "ARRAY<INT64>");
+    builder.addColumn("test", "arrfloat64", "ARRAY<FLOAT64>");
+    builder.addColumn("test", "arrstring", "ARRAY<STRING>");
+    builder.addColumn("test", "arrbytes", "ARRAY<BYTES>");
+    builder.addColumn("test", "arrtimestamp", "ARRAY<TIMESTAMP>");
+    builder.addColumn("test", "arrdate", "ARRAY<DATE>");
+
+    builder.addColumn("test", "nullarrbool", "ARRAY<BOOL>");
+    builder.addColumn("test", "nullarrint64", "ARRAY<INT64>");
+    builder.addColumn("test", "nullarrfloat64", "ARRAY<FLOAT64>");
+    builder.addColumn("test", "nullarrstring", "ARRAY<STRING>");
+    builder.addColumn("test", "nullarrbytes", "ARRAY<BYTES>");
+    builder.addColumn("test", "nullarrtimestamp", "ARRAY<TIMESTAMP>");
+    builder.addColumn("test", "nullarrdate", "ARRAY<DATE>");
+
+    allTypesSchema = builder.build();
+  }
+
+  @Test
+  public void testAllTypes() throws Exception {
+    encodeAndVerify(g(appendAllTypes(Mutation.newInsertOrUpdateBuilder("test")).build()));
+    encodeAndVerify(g(appendAllTypes(Mutation.newInsertBuilder("test")).build()));
+    encodeAndVerify(g(appendAllTypes(Mutation.newUpdateBuilder("test")).build()));
+    encodeAndVerify(g(appendAllTypes(Mutation.newReplaceBuilder("test")).build()));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testUnknownColumn() throws Exception {
+    SpannerSchema.Builder builder = SpannerSchema.builder();
+    builder.addKeyPart("test", "bool_field", false);
+    builder.addColumn("test", "bool_field", "BOOL");
+    SpannerSchema schema = builder.build();
+
+    Mutation mutation = Mutation.newInsertBuilder("test").set("unknown")
+        .to(true).build();
+    encodeAndVerify(g(mutation), schema);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
```

Comment:
use `@Rule ExpectedException` instead, and specify more precise expectations on the exception.

Reply:
Done